client/src/Pages/TripDetails.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect } from 'react';
+import { Button } from '../components/ui/button';
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from '../components/ui/popover';
+import { AiOutlinePlusCircle, AiOutlineMinusCircle } from 'react-icons/ai';
+import { FaMapLocationDot } from 'react-icons/fa6';
+import { useParams } from 'react-router-dom';
+import { TripEntry, icons, updateTrip } from '../lib/data';
+import useFindTrip from '../hooks/useFindTrip';
+
+type TripProps = {
+  onClick: (trip: TripEntry) => void;
+};
+
+export default function TripDetails({ onClick }: TripProps) {
+  const [activeIcon, setActiveIcon] = useState('');
+  const { tripId } = useParams();
+  const trip = useFindTrip(1, Number(tripId));
+  const { tripName, startDate, endDate, iconUrl } = trip;
+
+  useEffect(() => {
+    setActiveIcon(iconUrl);
+  }, [iconUrl]);
+
+  function handleIconChange(icon: string) {
+    setActiveIcon(icon);
+    updateTrip(trip, 1, trip.tripId, icon);
```

Comment:
You need to `await` this and put it in a try/catch so you can display any errors to the user.